src/pages/SignInAdminPage/SignInAdminForm/index.tsx

```diff
@@ -0,0 +1,191 @@
+import { useForm, Controller, SubmitHandler } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+import { Typography } from '@mui/material';
+import { Box } from '@mui/system';
+import { FetchBaseQueryError } from '@reduxjs/toolkit/query';
+import { useLoginMutation } from 'src/redux/auth/authApi';
+import { useAppDispatch } from 'src/redux/auth/hooks/hooks';
+import {
+  loginStart,
+  loginSuccess,
+  loginFailure,
+  setTokens,
+  setUser,
+} from 'src/redux/auth/authSlice';
+import { ILoginDto, ILoginResponse } from 'src/redux/auth/types/user';
+import theme from 'src/theme';
+import { appErrors, urls, validations } from 'src/common/constants';
+import PasswordInput from 'src/components/shared/PasswordInput';
+import {
+  InputPaddingVariants,
+  InputStyleVariants,
+} from 'src/components/shared/StyledInput/types';
+import StyledInput from 'src/components/shared/StyledInput';
+import StyledButton from 'src/components/shared/StyledButton';
+import {
+  PaddingVariants,
+  StyleVariants,
+} from 'src/components/shared/StyledButton/types';
+import LabelText from 'src/components/shared/LabelText';
+import TitleInputWrapper from 'src/components/shared/TitleInputWrapper';
+import useToast from 'src/components/shared/toasts/components/ToastProvider/ToastProviderHooks';
+import { SerializedError } from 'src/redux/user/types';
+import FormStyled, { ErrorMessage } from './styles';
+
+interface IFormInput {
+  email: string;
+  password: string;
+}
+
+function isFetchBaseQueryError(error: unknown): error is FetchBaseQueryError {
+  return typeof error === 'object' && error != null && 'status' in error;
+}
+
+function isSerializedError(error: unknown): error is SerializedError {
+  return typeof error === 'object' && error != null && 'message' in error;
+}
```

Comment:
I assume it should be used for any component? or a lot of them?

Reply:
Yes, I changed the way get error message so now this part doesn't repeat itself.